client/src/pages/CartPage/CartPage.js

```diff
@@ -1,54 +1,130 @@
-import { memo, useEffect, useMemo, useState } from "react";
-import { Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from '@material-ui/core';
+import { memo, useCallback, useEffect, useMemo, useState } from "react";
+import { Button, Paper, Table, TextField, TableBody, TableCell, TableContainer, TableFooter, TableHead, TableRow, Typography, CircularProgress } from '@material-ui/core';
+import { useDispatch, useSelector } from 'react-redux';
+import CartTableItem from "./CartTableItem";
+import { transferCurrency } from "../../functions";
+import { useStyles } from "./CartPageClasses";
 import { useHttp } from "../../hooks/http.hook";
-import { useSelector } from 'react-redux';
 import { POST } from "../../constants";
-import CartTableItem from "./CartTableItem";
+import { clearProductCart } from "../../redux/reducers/appShopCardReducer";
 
 function CartPage() {
-  const [products, setProducts] = useState([]);
+  const classes = useStyles();
+
+  const { request } = useHttp();
+
+  const dispatch = useDispatch();
 
-  const { request, status } = useHttp();
+  const [values, setValues] = useState({
+    text: '',
+    adress: ''
+  });
 
-  const { isAuth, shopCardProducts, currency } = useSelector(store => ({
-    isAuth: store.appAuth.isAuth,
-    shopCardProducts: store.appShopCard.products,
-    currency: store.appCurrency
+  const handleInputChange = useCallback(event => {
+    setValues(prevValues => ({
+      ...prevValues,
+      [event.target.name]: event.target.value
+    }));
+  }, []);
+
+  const { products, currency, productsStatus } = useSelector(store => ({
+    products: store.appShopCard.products,
+    currency: store.appCurrency,
+    productsStatus: store.appUser.status
   }));
-  
-  const productIds = useMemo(() => shopCardProducts.map(product => product.id), [shopCardProducts]);
+
+  const productsCount = useMemo(() => {
+    return products.reduce((accum, currProd) => accum + currProd.count, 0);
+  }, [products]);
+
+  const productsPrice = useMemo(() => {
```

Comment:
It's better to extract such helper functions into selectors, and later you will be able to reuse this logic